From 36094a0ce47327db46bea6705f78d1a4691e2110 Mon Sep 17 00:00:00 2001
From: Lei Huang <Lei.Huang@amd.com>
Date: Tue, 4 Jul 2023 10:24:55 +0800
Subject: [PATCH 449/458] xdg-shell: update xdg version for dpms

[Why]
we have update wayland protocols version for add dpms

[How]
need also update service xdg version

Change-Id: I39ce3441f3bf7cad47eb2f09bb58dd80f0f9cc6d
---
 src/video/wayland/SDL_waylandvideo.c | 2 +-
 wayland-protocols/xdg-shell.xml      | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/video/wayland/SDL_waylandvideo.c b/src/video/wayland/SDL_waylandvideo.c
index 5b4a358ee..5bbaa6fd1 100644
--- a/src/video/wayland/SDL_waylandvideo.c
+++ b/src/video/wayland/SDL_waylandvideo.c
@@ -834,7 +834,7 @@ static void display_handle_global(void *data, struct wl_registry *registry, uint
     } else if (SDL_strcmp(interface, "wl_seat") == 0) {
         Wayland_display_add_input(d, id, version);
     } else if (SDL_strcmp(interface, "xdg_wm_base") == 0) {
-        d->shell.xdg = wl_registry_bind(d->registry, id, &xdg_wm_base_interface, SDL_min(version, 3));
+        d->shell.xdg = wl_registry_bind(d->registry, id, &xdg_wm_base_interface, SDL_min(version, 4));
         xdg_wm_base_add_listener(d->shell.xdg, &shell_listener_xdg, NULL);
     } else if (SDL_strcmp(interface, "wl_shm") == 0) {
         d->shm = wl_registry_bind(registry, id, &wl_shm_interface, 1);
diff --git a/wayland-protocols/xdg-shell.xml b/wayland-protocols/xdg-shell.xml
index 368590bc6..792d5ae41 100644
--- a/wayland-protocols/xdg-shell.xml
+++ b/wayland-protocols/xdg-shell.xml
@@ -29,7 +29,7 @@
     DEALINGS IN THE SOFTWARE.
   </copyright>
 
-  <interface name="xdg_wm_base" version="3">
+  <interface name="xdg_wm_base" version="4">
     <description summary="create desktop-style surfaces">
       The xdg_wm_base interface is exposed as a global object enabling clients
       to turn their wl_surfaces into windows in a desktop environment. It
-- 
2.17.1

